tests/Contrib/Unit/OTLPGrpcSpanConverterTest.php

```diff
@@ -0,0 +1,209 @@
+        $span->setAttribute('integer-1', 1024);
+        $span->setAttribute('integer-2', 0);
+        $span->setAttribute('float', '1.2345');
+        $span->setAttribute('boolean-1', true);
+        $span->setAttribute('boolean-2', false);
+        $span->setAttribute('list-of-strings', $listOfStrings);
+        $span->setAttribute('list-of-numbers', $listOfNumbers);
+        $span->setAttribute('list-of-booleans', $listOfBooleans);
+        $span->setAttribute('list-of-random', $listOfRandoms);
+
+        $converter = new SpanConverter('test.name');
+        $tags = $converter->as_otlp_span($span)->getAttributes();
+
+        // // Check that we can convert all attributes to tags
+        $this->assertCount(10, $tags);
+
+        // // Tags destined for Otlp must be pairs of strings
+        // foreach ($tags as $tagKey => $tagValue) {
+        //     $this->assertIsString($tagKey);
+        //     $this->assertIsString($tagValue);
+        // }
+
+        // $this->assertEquals($tags['string'], 'string');
+        // $this->assertEquals($tags['integer-1'], '1024');
+        // $this->assertEquals($tags['integer-2'], '0');
+        // $this->assertEquals($tags['float'], '1.2345');
+        // $this->assertEquals($tags['boolean-1'], 'true');
+        // $this->assertEquals($tags['boolean-2'], 'false');
+
+        // // Lists must be casted to strings and joined with a separator
+        // $this->assertEquals($tags['list-of-strings'], join(',', $listOfStrings));
+        // $this->assertEquals($tags['list-of-numbers'], join(',', $listOfNumbers));
+        // $this->assertEquals($tags['list-of-booleans'], 'true,true,false,true');
+
+        // // This currently works, but OpenTelemetry\Trace\Span should stop arrays
+        // // containing multiple value types from being passed to the Exporter.
+        // $this->assertEquals($tags['list-of-random'], 'true,1,2,3,false,string-1,3.1415');
+    }
+
+    public function testOtlpHappyPathSpan()
+    {
+        $start_time = 1617315085034507008;
+        $end_time = 1617313804325769988;
+
+        // Construct a comprehensive happy path Span in the SDK
+        $sdk = new Span(
+            'http_get',
+            new SpanContext(
+                bin2hex('0000000000000001'), // traceId
+                bin2hex('00000001'), // spanId
+                0, // traceFlags
+            ),
+            null, // parentSpanContext
+            null, // sampler
+            ResourceInfo::create(
+                new Attributes([
+                    'instance' => 'test-a',
+                ])
+            )
+        );
+
+        $sdk->setStartEpochTimestamp($start_time);
+
+        $sdk->setAttribute('user', 'alice');
+        $sdk->setAttribute('authenticated', true);
+
+        $sdk->addEvent('Event1', 1617313804325769955, new Attributes(['sucess' => 'yes']));
+
+        $sdk->setSpanStatus(SpanStatus::OK);
+
+        $sdk->end($end_time); // Setting the end timestamp does not work how I expect it to
```

Comment:
This is the line that I'm struggling to make the test pass. /cc @bobstrecansky

1) Tried passing in a timestamp
2) Tried not passing a timestamp, but fetching it later with `$sdk->getEnd()` and passing this to the Proto Span didn't work either.

I've ran into complications with setting an end time/duration when trying to write database instrumentation in Laravel too.

Reply:
I spoke with Sean about this in DM, I think we can use:

`Clock::get())->timestamp()`

To accomplish his timing demands here.